Source/JavaScriptCore/dfg/DFGSpeculativeJIT.cpp

```diff
@@ -16748,6 +16748,53 @@ void SpeculativeJIT::compileNumberIsNaN(Node* node)
     }
 }
 
+void SpeculativeJIT::compileNumberIsFinite(Node* node)
+{
+    switch (node->child1().useKind()) {
+    case DoubleRepUse: {
+        SpeculateDoubleOperand argument(this, node->child1());
+        GPRTemporary scratch(this);
+        FPRTemporary diff(this);
+        FPRTemporary neg(this);
+
+        FPRReg argumentFPR = argument.fpr();
+        GPRReg scratchGPR = scratch.gpr();
+        FPRReg diffFPR = diff.fpr();
+        FPRReg negFPR = neg.fpr();
+
+        negateDouble(argumentFPR, negFPR);
+        addDouble(argumentFPR, negFPR, diffFPR);
```

Comment:
Yeah, if we can just use `subDouble` here, that's better.
```
subDouble(argumentFPR, argumentFPR, diffFPR);
```

Reply:
And also can you add a test? It seems that they are missed in the test.

Reply:
Added `LayoutTests/js/script-tests/Number-isFinite.js`